feat(sistent): add Table to components documentation

naikmubashir opened the pull request:
**Description**
This PR adds comprehensive documentation for the Table component (ResponsiveDataTable) to the Sistent components documentation page at `layer5.io/projects/sistent/components`. The documentation follows the established patterns used by other components like Button and includes complete examples, guidance, and code references.

### Changes Made:
- src/sections/Projects/Sistent/components/table/
	- [x] **index.js** # Overview page with basic examples
	- [x] **guidance.js** # Design guidelines and best practices
	- [x] **code.js** # Comprehensive code examples
- [x] src/sections/Projects/**content.js**
- [x] src/components/SistentNavigation/**content.js**

This PR fixes #5962 

**Notes for Reviewers**


**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.


<!--
Thank you for contributing to Layer5 projects! 

Contributing Conventions:

1. Include descriptive PR titles with [<component-name>] prepended.
2. Build and test your changes before submitting a PR. 
3. Sign your commits

By following the community's contribution conventions upfront, the review process will 
be accelerated and your PR merged more quickly.
-->

naikmubashir committed:
feat(sistent): add Table to SistentNavigation

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
feat(sistent): add Table to component documentation

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
feat(sistent): add Table usage guidance and best practices

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
feat(sistent): add Table interative code examples

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
feat(sistent): add Table to components data list

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

Rajesh-Nagarajan-11 reviewed (commented):
atleast add these too(in guidance) @naikmubashir  :

- What are options values...
- How to enable column sort
- what are multiselect and none select
- disabled row
- sub tables

naikmubashir committed:
Merge branch 'master' into master

Signed-off-by: Naik Mubashir <124430100+naikmubashir@users.noreply.github.com>

naikmubashir committed:
fix(sistent): fix eslint errors in SistentNavigation

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

l5io commented:
🚀 Preview for commit 11a1617976f45a569a12747ff08daee73ccd8338 at: https://68f875ebee2407df2194c8ea--layer5.netlify.app

Rajesh-Nagarajan-11 commented:
Thank you for your contribution!
Let's discuss this during the website call today at 5:30 PM IST | 7 AM CT
Add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk), if you would 😄

naikmubashir committed:
merge new changes

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
fix(sistent): fix new updates

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
Merge branch 'master' of https://github.com/layer5io/layer5

Rajesh-Nagarajan-11 commented:
Thank you for your contribution!
Let’s discuss this during the website call tomorrow (November 3) at 5:30 PM IST | 7:00 AM CT.
Please add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk)
 😊

naikmubashir committed:
fix: resolve merge conflict

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
Merge branch 'master' of https://github.com/layer5io/layer5

naikmubashir committed:
fix: update code.js

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

naikmubashir committed:
fix(sistent): prevent SSR issues with Table component rendering

Added client-side mounting check to ResponsiveDataTable components to fix
Gatsby static HTML build failure. The MUI theme.alpha function was causing
errors during SSR. Tables now only render after component mounts in browser.

Signed-off-by: Naik Mubashir <naikmubashir095@gmail.com>

Rajesh-Nagarajan-11 committed:
Merge branch 'master' into master

leecalcote commented:
Merge conflicts...

Fireentity commented:
Thank you for your contribution!
Let’s discuss this during the website call today (December 1st) at 8:00AM CT | 6:30PM IST.
Please add it as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk)
😊

naikmubashir committed:
Merge branch 'master' into master

Signed-off-by: Naik Mubashir <124430100+naikmubashir@users.noreply.github.com>

leecalcote committed:
Merge branch 'master' into master

l5io committed:
Updated feature data from spreadsheet

Signed-off-by: l5io <ci@layer5.io>

naikmubashir commented:
@kishore08-07  Can we just reopen this pr. I would to work on this further 
@Rajesh-Nagarajan-11

saurabhraghuvanshii committed:
Merge branch 'master' into master

naikmubashir committed:
Merge branch 'master' into master

Signed-off-by: Naik Mubashir <124430100+naikmubashir@users.noreply.github.com>

l5io committed:
Updated feature data from spreadsheet

Signed-off-by: l5io <ci@layer5.io>

l5io committed:
Updated feature data from spreadsheet

Signed-off-by: l5io <ci@layer5.io>

rishiraj38 reviewed (commented):
Thanks @naikmubashir, could you please take a look at the updated version of this issue? #7607
